h/security/identity.py

```diff
@@ -94,3 +94,11 @@ def from_models(cls, user: User = None, auth_client: AuthClient = None):
                 LongLivedAuthClient.from_model(auth_client) if auth_client else None
             ),
         )
+
+    @staticmethod
+    def authenticated_userid(identity: Self | None) -> str | None:
+        """Return the authenticated_userid from the given identity."""
+        if identity and identity.user:
+            return identity.user.userid
+
+        return None
```

Comment:
I moved the `IdentityBasedPolicy.authenticated_userid()` method to a static method on the `Identity` class itself. Alternatively this could just be a standalone helper function.